nodebuilder/header/service.go

```diff
@@ -8,29 +8,7 @@ import (
 	"github.com/celestiaorg/celestia-node/header/sync"
```

Comment:
Should we move this file's content to `header/header.go`? The issue seems to indicate that `service.go` should be deprecated in favor of `pkg.go`.

Reply:
This is an exception because the `nodebuilder/header` package actually contains the implementation of the "header service" (which just calls underlying services). So `service.go` is this implementation, and `header.go` is the `API`/`Module` definition as it is in the other `nodebuilder` packages